vortex-array/src/array/sparse/variants.rs

```diff
@@ -48,6 +48,20 @@ impl ArrayVariants for SparseArray {
 impl NullArrayTrait for SparseArray {}
 
 impl BoolArrayTrait for SparseArray {
+    fn invert(&self) -> VortexResult<Array> {
+        SparseArray::try_new(
+            self.indices(),
+            self.values().with_dyn(|a| {
+                a.as_bool_array()
+                    .ok_or_else(|| vortex_err!("Not a bool array"))
+                    .and_then(|b| b.invert())
+            })?,
+            self.len(),
+            self.fill_value().clone(),
```

Comment:
there's some interesting semantics here whether this should invert non null fill value

Reply:
rereading: yes, i think this should invert the non-null fill value?

Otherwise, not(canonical(sparsearray)) != canonical(not(sparsearray)) right?

Reply:
yes, you're right